### ISSUE TITLE
can you provide this data?

### ISSUE THREAD

**Issue by LDJ20210926** [severi-rissanen/cevae#1]
![image](https://user-images.githubusercontent.com/91396555/167579848-d692425f-3752-4dfe-b0f8-490e4bd39d4d.png)
please

**Comment by severi-rissanen** [comment]
Yes! You can find it over here: https://www.fredjo.com/files/ihdp_npci_1-1000.train.npz.zip.

**Comment by LDJ20210926** [comment]
first, thanks for you very much ! can I ask you some questions?  I find your CEVAE.py , or in other words, the decoder or incoder, that you may never consider the AUXILIARY LOSS from the paper, what's your idea about it?

**Comment by LDJ20210926** [comment]
![image](https://user-images.githubusercontent.com/91396555/169238226-6141ba4b-418e-46c9-a0e5-184438a4cba3.png)
I don't understand the meaning of yours, can your answer me?

**Comment by severi-rissanen** [comment]
Hi, sorry for not answering earlier! For some reason I didn't get a notification. The auxiliary network (and corresponding auxiliary loss) was meant for estimation of individual treatment effects using Eq.1 in the paper. I didn't use it because I was only interested in average treatment effects. In principle, it is completely disconnected from the actual VAE during training, so you could train it as a separate neural network and then use Eq.1 and the method at the second paragraph of the Experiments-section. 

I'm not sure what is the context in the image. Visualization of a toy binary data set? Was this from the official code release (https://github.com/severi-rissanen/critical_look_causal_dlvms) or here?

**Comment by LDJ20210926** [comment]
firstly, thanks for your responsing, the sencond question is that from the context of the below picture
![image](https://user-images.githubusercontent.com/91396555/169484725-d793183e-7949-478e-abb5-36670bffb64e.png)

**Comment by severi-rissanen** [comment]
Ah alright, I think that file corresponds to the toy experiment in the original CEVAE paper. Have you just tried running it and seeing what kind of plot it outputs? I think the point of having a different y value on the plot is to signify the difference in z values (so y in that piece of code is not the same as y in CEVAE, instead it's just the plotting y). 

I think that I got that particular piece of code from somewhere else, and didn't use it much myself.

**Comment by LDJ20210926** [comment]
ohyeah, thinks for your answering, and mybe, my questions  are  lots, but I really want to ask you about how to learning parameters

**Comment by severi-rissanen** [comment]
Yeah no problem, which parameters do you mean?

**Comment by LDJ20210926** [comment]
Yeah,thank you very much, I just got it. And another one, I really want to ask you if I use q(t|z) and q(y|z) instead of the equils in paper about auxiliary parts. Do you think it's reasonable?

**Comment by LDJ20210926** [comment]
![image](https://user-images.githubusercontent.com/91396555/169971301-1a465815-b12b-4735-a932-c27222f9eee0.png)
can you explain why do this, if z is binrary, can you tell me more detail? And if z is disentangled to z1,z2, how can I design the z binary loss about q_z_probs
from 
https://github.com/severi-rissanen/cevae/blob/master/cevaetools.py

**Comment by severi-rissanen** [comment]
Yeah I think that if you're going to use the auxiliary loss, you should use the equations in the paper. But: You should be a bit vary about the results, since our paper did show that the model is not always reliable. 

qz_probs contains the probabilities of z=0 and z=1. You can always get one from the other with prob(z=1) = 1 - prob(z=0), but they are both in qz_probs for convenience over there. Later on they are braodcased to calculate the expected value of x_loss.

**Comment by LDJ20210926** [comment]
yeah, i got it, thank you very much, And I want to ask you about the result of  jobs dataset, i never see it in the code

**Comment by LDJ20210926** [comment]
Can I ask you about the latented variables, if Z is Bernoulli distribution，then how can I use reparameterization? or in orther words, Can I calculate the gradient, please!

**Comment by severi-rissanen** [comment]
Yeah, I think I didn't experiment with the jobs data set, it was only in the original cevae paper. IHDP sort of served the same thing for us, but maybe you can find info from the original paper or others that have used it?

And that's a good question! The thing is that you don't need the reparametrization trick at all, you can calculate the expectation value exactly! So you do two passes through the decoder with z=0 and z=1, and then weight the results with the probabilities for z=0 and z=1. If z is high-dimensional, you can't evaluate the expectation exactl, but in this special case it is possible.

**Comment by severi-rissanen** [comment]
So you do get the gradient trivially as well

**Comment by LDJ20210926** [comment]
but if I have the category variables(not just binary) for z, there are any methods for it

**Comment by LDJ20210926** [comment]
Can I ask you how can I split train/test/valid for datasets and design the network

**Comment by severi-rissanen** [comment]
Yeah, you can try to do the same thing even with a more complicated categorical variable, but this is not very scalable, of course. I think there was this Gumbel-softmax trick that sort of does the reparametrization for categorical variables: https://arxiv.org/pdf/1611.01144.pdf 

The train/validation/test splits you can choose however you want. For the paper, I just used a few fully connected layers with ELU activation functions. Mainly the thing is that you need to have enough capacity to work with the data in question (so the network has to be wide / deep enough). Of course this then depends on the type of data you have.

**Comment by severi-rissanen** [comment]
Yeah no it doesn't need to be super big. It depends entirely on the data, you can just use the designs from the paper if you want. As in general in ML, you don't want to underfit or overfit.

**Comment by LDJ20210926** [comment]
Can I ask you if I design the toy data as follow is reasonable?
![image](https://user-images.githubusercontent.com/91396555/171892222-962dc50c-6356-4a3d-8257-1eaf63b384e5.png)

**Comment by LDJ20210926** [comment]
Can I ask your why the error is so large in below from the link:
https://github.com/severi-rissanen/cevae/blob/master/counterfactual_inference_experiment_suite.ipynb
![image](https://user-images.githubusercontent.com/91396555/171991310-b9144c98-896e-4b3a-825b-4fde14708f4a.png)

**Comment by severi-rissanen** [comment]
That seems reasonable, although realistically you will probably need at least two proxies x. Good idea to check how large is the true ATE (if it is close to 1 in the true data generating process, the problem is overly easy since an estimator that always outputs just ~1 is already correct, without much need for machine learning. (I think this was the case with the parameters in the original paper)

Not sure what those numbers are, that is from an earlier piece of code that wasn't mine and I didn't use it much.

**Comment by LDJ20210926** [comment]
I'm very sorry, I couldn't log in to GitHub due to a problem with the vpn before. And I want to ask you about the scoure of the link
https://github.com/severi-rissanen/cevae/blob/master/lineartoydata.py.And why desgin

**Comment by LDJ20210926** [comment]
And I want to ask you what's the mean of t_a, t_b, or how to define them ?
![image](https://user-images.githubusercontent.com/91396555/174827640-ec5d216c-f759-4549-b6ed-c649e26f557c.png)

**Comment by severi-rissanen** [comment]
Hi, it seems that you didn't write the question in full here. The lineartoydata.py seems to be my code, and can produce the linear-Gaussian data in our paper. The meanings of t_a and t_b are that they are the logistic regression coefficients that are used to generate the binary t data from the continuous-valued z.

**Comment by LDJ20210926** [comment]
yeah, I find your paper, and know it , thank you very much!
